mullvad-relay-selector/src/relay_selector/mod.rs

```diff
@@ -0,0 +1,859 @@
+//! The implementation of the relay selector.
+
+mod detailer;
+mod helpers;
+mod matcher;
+mod parsed_relays;
+pub mod query;
+
+use chrono::{DateTime, Local};
+use itertools::Itertools;
+use once_cell::sync::Lazy;
+use std::{
+    path::Path,
+    sync::{Arc, Mutex},
+    time::SystemTime,
+};
+
+use matcher::{BridgeMatcher, RelayMatcher, WireguardMatcher};
+use mullvad_types::{
+    constraints::Constraint,
+    custom_list::CustomListsSettings,
+    endpoint::{MullvadEndpoint, MullvadWireguardEndpoint},
+    location::{Coordinates, Location},
+    relay_constraints::{
+        BridgeSettings, BridgeState, InternalBridgeConstraints, ObfuscationSettings,
+        OpenVpnConstraints, RelayOverride, RelaySettings, ResolvedBridgeSettings,
+        SelectedObfuscation, WireguardConstraints,
+    },
+    relay_list::{Relay, RelayList},
+    settings::Settings,
+    CustomTunnelEndpoint,
+};
+use talpid_types::{
+    net::{obfuscation::ObfuscatorConfig, proxy::CustomProxy, TransportProtocol, TunnelType},
+    ErrorExt,
+};
+
+use crate::error::Error;
+
+use self::{
+    detailer::{OpenVpnDetailer, WireguardDetailer},
+    matcher::AnyTunnelMatcher,
+    parsed_relays::ParsedRelays,
+    query::{BridgeQuery, Intersection, OpenVpnRelayQuery, RelayQuery, WireguardRelayQuery},
+};
+
+/// [`RETRY_ORDER`] defines an ordered set of relay parameters which the relay selector should prioritize on
```

Comment:
I suggest a brief clarification that there queries can never overwrite the user settings and are only chose if they are compatible according to the `Intersection` trait

Reply:
Done!